src/evilvte.1
.TH EVILVTE 1 "March  8, 2008"
.SH NAME
evilvte \- an VTE based super lightweight terminal emulator
.SH SYNOPSIS
.B evilvte
.RI [ options ]
.SH DESCRIPTION
.B evilvte
is a terminal emulator. It supports almost everything VTE provides.
It also supports tabs, tabbar autohide, and switch encoding at runtime.
Configuration is via editing source code and recompilation.
All arguments in this manpage are also optional.
A standard stripped binary size of evilvte is less than
.B 9 kilobytes.
.SH OPTIONS
.TP
.B \-e [ program ]
Specifies the program to be run in the evilvte window. This option may be disabled at build-time.
.TP
.B \-v
Show version of evilvte. This option may be disabled at build-time.
.SH AUTHOR
evilvte was written by Wen-Yen Chuang <caleb@calno.com>


ChangeLog
UNRELEASED
 * Make close button smaller
 * Improve resizing behavior with "-VteTerminal-inner-border" property

0.5.1~pre1

 * Temporary workaround for "-VteTerminal-inner-border" property
 * Mouse clicking can also turn off urgency hint now

0.5.0

 The "One binary to rule them all!" Release

 * Fix a bug for -e option
 * Fix a bug for input method menu
 * New configure option: --with-gtk=2+3|3+2
   The binary will not link to VTE nor GTK+, thus can support GTK+ 2.x and 3.x
   at the same time. It still depends on VTE runtime, of course.
   Use --2 or --3 command line options to choose the version of GTK+.
   If evilvte segfaults with --3 option, please try export USE_ACCESSIBLE=0 or
   USE_ACCESSIBLE=1.
 * Drop codes with VTE version < 0.17.1

0.5.0~pre1
 * Sync COLOR_STYLE: ZENBURN with xterm
   Thanks to Artur Dryomov <me AT ming13 DOT ru>
   Previous ZENBURN is renamed to ZENBURN_DARK
 * Replace all deprecated functions that have suitable replacements
 * Make everything GTK+ 3.1.92 compatible
   Use GTK_FONT_CHOOSER_DIALOG only for GTK+ version >= 3.1.90
 * Make everything VTE 0.29.1 compatible
   If you encounter problems, try export CFLAGS="-DUSE_ACCESSIBLE=0"
   and rebuild evilvte.

0.4.9
 * Fix makefile parallelization
   Thanks to Thomas Weißschuh <thomas_weissschuh AT lavabit DOT com>
 * Use GTK_FONT_CHOOSER_DIALOG for GTK+ version >= 3.1.12
 * Expand VTE widget in GtkGrid

0.4.9~pre5
 * Support GtkGrid
 * Fix an SCROLLBAR minor bug with option OFF_L and OFF_R
 * New icon set under the license:
   GNU General Public License version 2 or any later version
   Thanks to Alexander Inglessi <inglessi AT gmail DOT com>
 * Use vte_terminal_fork_command() by default. Please see:
   http://evilvte.blogspot.com/2011/03/issue-evilvte-048-makes-zsh-segfault.html

0.4.9~pre4
 * Support SEARCH_CASE_SENSITIVE
   I will make it toggleable at runtime in the future.
   But I am lazy to write it now. :-P
   Patches are very welcome.
 * Make everything GLib 2.29.8 compatible
 * Sign SHA1SUMS using RSA key ID BE8BA819 and DSA key ID E98D6159

0.4.9~pre3
 * Grayed out menu items according to their status:
   "Copy", "Paste", "Zoom default"

0.4.9~pre2
 * Minor tweaks in TABBAR_AUTOHIDE for GTK+ 3.x
 * Hide VTE_FIXED color style which was merged in VTE version >= 0.17.1
 * Drop !TAB codes since evilvte is GtkNotebook-only for a very long time.
   For those who do not use tabs, the binary size will increase a little.

0.4.9~pre1
 * Support "Auto-click" when there is only one effective menu item.
   You can assign anything to right-click now.
 * Support embedded GTK3_CSS

0.4.8

 The "Welcome, GTK+ 3.0.0!" Release

 * Better compatibility with string matching
 * Fix a bug that made MATCH_STRING fail when MENU is not enabled
   Thanks to Kabakov Boris <pigritor AT gmail DOT com>
 * New COLOR_STYLE option: ZENBURN
   Thanks to thagraal and <ezrgzx AT gmail DOT com>
 * New options: MATCH_STRING_HTTP
                MATCH_STRING_MAIL
                MATCH_STRING_FILE
 * Hide MATCH_STRING
 * Make everything GTK+ 3.0.0 compatible

0.4.8~pre3
 * Fix a bug for multiple -e options in command line
 * Fix an i18n bug when linked to GTK+ version >= 2.90.0

0.4.8~pre2
 * Initial support for vte_terminal_search_*
   The feature of VTE is still quite buggy.
   Please see: https://bugzilla.gnome.org/627886
 * Support underscore in LABEL_* to indicate mnemonic of menu item
 * Drop MATCH_STRING_EXEC which was deprecated since 0.4.7~pre1
 * Hide DEFAULT_EMULATION_TYPE, TAB_BORDER_HORIZONTAL and TAB_BORDER_VERTICAL
 * Provide a clearsigned SHA1SUMS using DSA key ID E98D6159

0.4.8~pre1
 * Status bar and scrollbar can be hidden initially and be toggled on later
 * Better behavior for *_TOGGLE_TABBAR with TABBAR_AUTOHIDE enabled
 * Sync window state with window manager
 * Fix some cross-platform issues
   Thanks to Aragon Gouveia <aragon AT phat DOT za DOT net>
 * Initial support for new resizing behavior of VTE
   If you are using GTK+ version >= 2.91.1 and VTE version >= 0.27.1,
   the window size of evilvte is not always correct currently.

0.4.7.1
 * Fix a segfault bug of HOTKEY_MIMIC_SCROLL_UP and HOTKEY_MIMIC_SCROLL_DOWN

0.4.7
 * Better compatibility with WM_CLASS
   - Accept environment variable RESOURCE_NAME
   - Document --name and --class command line options in manpage
   - Note that -title and -T no longer changes WM_CLASS
   - Thanks to Zveroy <zveroy AT gmail DOT com> and
               Tetralet <tetralet AT gmail DOT com>
 * Accept environment variable PKG_CONFIG_PATH, thanks to letoh
 * Support -g +X+Y command line option to assign initial window geometry
 * Prevent tab labels getting focus

0.4.7~pre5
 * Support -fn command line option to assign font and font size
 * Better compatibility with configure options

0.4.7~pre4
 * Support HOTKEY_MIMIC_SCROLL_*, thanks to qotsa
 * New configure option: --with-gtk=2.0|3.0

0.4.7~pre3
 * Support vte_terminal_fork_command_full
 * Support all positive integer font size
 * Drop gdk-pixbuf dependency. It was used for a bug only exists in VTE 0.24.0.

0.4.7~pre2
 * Support BELL_URGENT, aka WM_URGENT hint. Thanks to L29Ah.

0.4.7~pre1

 The "Thanks to Aragon!" Release

 * Rename MATCH_STRING_EXEC to MATCH_STRING_L
 * Provide MATCH_STRING_M
 * Fix two MATCH_STRING_* bugs
 * Fix two WINDOW_TITLE_DYNAMIC bugs
 * Fix a cross-platform issue

0.4.6
 * Workaround for VTE 0.24.0 vte_terminal_set_background_image_file() bug

0.4.6~pre6
 * Support CURSOR_SHAPE
 * New BACKSPACE_KEY and DELETE_KEY option: ERASE_TTY
 * Simplify Makefile and configure

0.4.6~pre5
 * Saturation can be adjusted by holding Ctrl and scrolling mouse up or down.
   Define both MOUSE_CTRL_SATURATION and HOTKEY to TRUE to use this function.
   If CTRL_ALT is TRUE, you have to hold Ctrl and Alt while scrolling.
 * Remove close button border in some specific themes
   Thanks to Tetralet <tetralet AT gmail DOT com>

0.4.6~pre4
 * Support MATCH_STRING_EXEC

src/evilvte.sh
#!/bin/sh
# Copyright (C) 2008-2009  Wen-Yen Chuang <caleb AT calno DOT com>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

NEWCONFFILE=src/evilvte.h

cat /dev/null > $NEWCONFFILE
rm -f $1 src/sakura.o

MENU_DEFAULT_ENCODING=`grep MENU_ENCODING_LIST src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | tail -n 1 | grep 'Default Encoding'`
if [ "$MENU_DEFAULT_ENCODING" != "" ]; then
  echo \#define MENU_DEFAULT_ENCODING 1 >> $NEWCONFFILE
fi

COLOR_STYLE_DEFINE=`grep COLOR_STYLE src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | tail -n 1 | awk '{print $3}'`
if [ "$COLOR_STYLE_DEFINE" = "VTE_FIXED" ]; then
  echo \#define COLOR_VTE_FIXED 1 >> $NEWCONFFILE
fi
if [ "$COLOR_STYLE_DEFINE" = "LINUX" ]; then
  echo \#define COLOR_LINUX 1 >> $NEWCONFFILE
fi
if [ "$COLOR_STYLE_DEFINE" = "RXVT" ]; then
  echo \#define COLOR_RXVT 1 >> $NEWCONFFILE
fi
if [ "$COLOR_STYLE_DEFINE" = "TANGO" ]; then
  echo \#define COLOR_TANGO 1 >> $NEWCONFFILE
fi
if [ "$COLOR_STYLE_DEFINE" = "XTERM" ]; then
  echo \#define COLOR_XTERM 1 >> $NEWCONFFILE
fi

DEFAULT_TERMINAL_SIZE_DEFINE=`grep DEFAULT_TERMINAL_SIZE src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | tail -n 1`
if [ "$DEFAULT_TERMINAL_SIZE_DEFINE" != "" ]; then
  COLUMNS_DEFINE=`echo $DEFAULT_TERMINAL_SIZE_DEFINE | awk '{print $3}' | cut -d x -f 1`
  ROWS_DEFINE=`echo $DEFAULT_TERMINAL_SIZE_DEFINE | grep x - | cut -d x -f 2`
  if [ "$ROWS_DEFINE" = "" ]; then
    ROWS_DEFINE=24
  fi
  echo \#define VTE_COLUMNS $COLUMNS_DEFINE >> $NEWCONFFILE
  echo \#define VTE_ROWS $ROWS_DEFINE >> $NEWCONFFILE
fi

TOGGLE_BG_ORDER_DEFINE=`grep TOGGLE_BG_ORDER src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | tail -n 1`

TOGGLE_BG_IMAGE=`echo $TOGGLE_BG_ORDER_DEFINE | grep Image`
if [ "$TOGGLE_BG_IMAGE" != "" ]; then
  echo \#define TOGGLE_BG_IMAGE 1 >> $NEWCONFFILE
fi

TOGGLE_BG_TRANSPARENT=`echo $TOGGLE_BG_ORDER_DEFINE | grep Transparent`
if [ "$TOGGLE_BG_TRANSPARENT" != "" ]; then
  echo \#define TOGGLE_BG_TRANSPARENT 1 >> $NEWCONFFILE
fi

TOGGLE_BG_NO_BACKGROUND=`echo $TOGGLE_BG_ORDER_DEFINE | grep 'No background'`
if [ "$TOGGLE_BG_NO_BACKGROUND" != "" ]; then
  echo \#define TOGGLE_BG_NO_BACKGROUND 1 >> $NEWCONFFILE
fi

TOGGLE_BG_OPACITY=`echo $TOGGLE_BG_ORDER_DEFINE | grep Opacity`
if [ "$TOGGLE_BG_OPACITY" != "" ]; then
  echo \#define TOGGLE_BG_OPACITY 1 >> $NEWCONFFILE
fi

MENU_CUSTOM_DEFINE=`grep MENU_CUSTOM src/custom.h | tr -s ' ' ' ' | sed 's/^ //' | grep -v ^\/\/ | tail -n 1`

MENU_COPY=`echo $MENU_CUSTOM_DEFINE | grep Copy`
if [ "$MENU_COPY" != "" ]; then
  echo \#define MENU_COPY 1 >> $NEWCONFFILE
fi

MENU_PASTE=`echo $MENU_CUSTOM_DEFINE | grep Paste`
if [ "$MENU_PASTE" != "" ]; then
  echo \#define MENU_PASTE 1 >> $NEWCONFFILE
fi

MENU_SELECT_ALL=`echo $MENU_CUSTOM_DEFINE | grep 'Select all'`
if [ "$MENU_SELECT_ALL" != "" ]; then
  echo \#define MENU_SELECT_ALL 1 >> $NEWCONFFILE
fi

MENU_COLOR_BACKGROUND=`echo $MENU_CUSTOM_DEFINE | grep 'Background tint'`
if [ "$MENU_COLOR_BACKGROUND" != "" ]; then
  echo \#define MENU_COLOR_BACKGROUND 1 >> $NEWCONFFILE
fi

MENU_TAB_ADD=`echo $MENU_CUSTOM_DEFINE | grep 'Add tab'`
if [ "$MENU_TAB_ADD" != "" ]; then
  echo \#define MENU_TAB_ADD 1 >> $NEWCONFFILE
fi

MENU_TAB_REMOVE=`echo $MENU_CUSTOM_DEFINE | grep 'Remove tab'`
if [ "$MENU_TAB_REMOVE" != "" ]; then
  echo \#define MENU_TAB_REMOVE 1 >> $NEWCONFFILE
fi

MENU_TAB_EDIT_LABEL=`echo $MENU_CUSTOM_DEFINE | grep 'Edit label'`
if [ "$MENU_TAB_EDIT_LABEL" != "" ]; then
  echo \#define MENU_TAB_EDIT_LABEL 1 >> $NEWCONFFILE
fi

MENU_OPEN_NEW_WINDOW=`echo $MENU_CUSTOM_DEFINE | grep 'New window'`
if [ "$MENU_OPEN_NEW_WINDOW" != "" ]; then
  echo \#define MENU_OPEN_NEW_WINDOW 1 >> $NEWCONFFILE
fi

MENU_QUIT=`echo $MENU_CUSTOM_DEFINE | grep Quit`
if [ "$MENU_QUIT" != "" ]; then
  echo \#define MENU_QUIT 1 >> $NEWCONFFILE
fi

MENU_FONT_BIGGER=`echo $MENU_CUSTOM_DEFINE | grep 'Zoom in'`
if [ "$MENU_FONT_BIGGER" != "" ]; then
  echo \#define MENU_FONT_BIGGER 1 >> $NEWCONFFILE
fi

MENU_FONT_SMALLER=`echo $MENU_CUSTOM_DEFINE | grep 'Zoom out'`
if [ "$MENU_FONT_SMALLER" != "" ]; then
  echo \#define MENU_FONT_SMALLER 1 >> $NEWCONFFILE
fi

MENU_FONT_DEFAULT_SIZE=`echo $MENU_CUSTOM_DEFINE | grep 'Zoom default'`
if [ "$MENU_FONT_DEFAULT_SIZE" != "" ]; then
  echo \#define MENU_FONT_DEFAULT_SIZE 1 >> $NEWCONFFILE
fi

MENU_RESET_TERMINAL=`echo $MENU_CUSTOM_DEFINE | grep 'Reset terminal'`
if [ "$MENU_RESET_TERMINAL" != "" ]; then
  echo \#define MENU_RESET_TERMINAL 1 >> $NEWCONFFILE
fi

MENU_RESET_AND_CLEAR=`echo $MENU_CUSTOM_DEFINE | grep 'Reset and clear'`
if [ "$MENU_RESET_AND_CLEAR" != "" ]; then
  echo \#define MENU_RESET_AND_CLEAR 1 >> $NEWCONFFILE
fi

MENU_FONT_SELECT=`echo $MENU_CUSTOM_DEFINE | grep 'Select font'`
if [ "$MENU_FONT_SELECT" != "" ]; then
  echo \#define MENU_FONT_SELECT 1 >> $NEWCONFFILE
fi

MENU_SEPARATOR=`echo $MENU_CUSTOM_DEFINE | grep Separator`
if [ "$MENU_SEPARATOR" != "" ]; then
  echo \#define MENU_SEPARATOR 1 >> $NEWCONFFILE
fi

SUBMENU_ENCODING_LIST=`echo $MENU_CUSTOM_DEFINE | grep 'Submenu encoding list'`
if [ "$SUBMENU_ENCODING_LIST" != "" ]; then
  echo \#define SUBMENU_ENCODING_LIST 1 >> $NEWCONFFILE
fi

SUBMENU_INPUT_METHOD=`echo $MENU_CUSTOM_DEFINE | grep 'Submenu input method'`
if [ "$SUBMENU_INPUT_METHOD" != "" ]; then
  echo \#define SUBMENU_INPUT_METHOD 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_BACKGROUND=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle background'`
if [ "$MENU_TOGGLE_BACKGROUND" != "" ]; then
  echo \#define MENU_TOGGLE_BACKGROUND 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_HOTKEYS=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle hotkeys locking'`
if [ "$MENU_TOGGLE_HOTKEYS" != "" ]; then
  echo \#define MENU_TOGGLE_HOTKEYS 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_ON_TOP=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle always on top'`
if [ "$MENU_TOGGLE_ON_TOP" != "" ]; then
  echo \#define MENU_TOGGLE_ON_TOP 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_SCROLLBAR=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle scrollbar'`
if [ "$MENU_TOGGLE_SCROLLBAR" != "" ]; then
  echo \#define MENU_TOGGLE_SCROLLBAR 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_STATUS_BAR=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle status bar'`
if [ "$MENU_TOGGLE_STATUS_BAR" != "" ]; then
  echo \#define MENU_TOGGLE_STATUS_BAR 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_TABBAR=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle tabbar'`
if [ "$MENU_TOGGLE_TABBAR" != "" ]; then
  echo \#define MENU_TOGGLE_TABBAR 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_DECORATED=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle window decorated'`
if [ "$MENU_TOGGLE_DECORATED" != "" ]; then
  echo \#define MENU_TOGGLE_DECORATED 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_FULLSCREEN=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle fullscreen'`
if [ "$MENU_TOGGLE_FULLSCREEN" != "" ]; then
  echo \#define MENU_TOGGLE_FULLSCREEN 1 >> $NEWCONFFILE
fi

MENU_TOGGLE_ANTI_ALIAS=`echo $MENU_CUSTOM_DEFINE | grep 'Toggle anti-alias'`
if [ "$MENU_TOGGLE_ANTI_ALIAS" != "" ]; then
  echo \#define MENU_TOGGLE_ANTI_ALIAS 1 >> $NEWCONFFILE
fi

MENU_CHANGE_SATURATION=`echo $MENU_CUSTOM_DEFINE | grep 'Adjust saturation'`
if [ "$MENU_CHANGE_SATURATION" != "" ]; then
  echo \#define MENU_CHANGE_SATURATION 1 >> $NEWCONFFILE
fi
